### ISSUE TITLE
Modify portsfetch to use git ports repository

### ISSUE THREAD

**Issue by marcpearson** [chriswells0/freebsd-scripts#3]
HI

Do you planning to modify portsfetch script to use the new git ports ?

**Comment by chriswells0** [comment]
Thanks for bringing this up! Let me know if you run into any issues with it.

**Comment by marcpearson** [comment]
if i can afford to give my vision

1. I think you should get the latest quarterly branch tag with something like this (This way checkout is not needed at first) :

git ls-remote --heads https://git.freebsd.org/ports.git | cut -f 2 | egrep "20[0-9]{2}Q[1-4]"| sed 's/refs\/heads\///g' | tail -1

2.  You should clone the desired branch only:

git clone --branch "$BRANCH" --single-branch "$URL" "$DESTDIR" 

3. If files exists at the $DESTDIR and user answer no, it's not logical to clone the ports with other files, i would do:

* ) echo "Nothing was done!"; exit 0 ;;

(FYI, portsnap is not compatible with the GIT ports tree, maybe later )

What do you think ?

**Comment by chriswells0** [comment]
Starting with 2) I thought that sounded like a good approach until I tried it. Cloning only the specific branch didn't save any disk space, which means there isn't any benefit that I'm aware of.

Here is a list of ports tree sizes:

```
756M    ports (managed by portsnap)
1.6G    ports-git (full repo clone)
1.6G    ports-git-branch (the command you provided)
1.6G    ports-git-branch-handbook (branch checkout per the handbook)
```

Plus, the _clone_ command you provided prevented me from switching to any other branch afterward. Unless I'm mistaken, that means I'd have to clone every time I changed branches rather than being able to pull updates and then switch. The command from the handbook doesn't have that limitation, but its disk usage was the same.

I would conclude that sticking with a full clone initially makes sense as it will be much faster to pull changes and then switch branches in the future. I'm open to hearing more if you have info I don't.

Then 1) This becomes moot if we clone the whole repo anyway.

And 3) It's really just a fallback to always try to update the existing ports tree. The effect is that someone who uses _quarterly_ on some machines and _latest_ on others can always run `portsfetch` to get the desired result without thinking about it.
